opsdroid/core.py

```diff
@@ -97,54 +107,95 @@ def critical(self, error, code):
         _LOGGER.critical(error)
         self.exit()
 
-    def call_stop(self):
-        """Signal handler to call disconnect and stop."""
-        future = asyncio.ensure_future(self.disconnect())
-        future.add_done_callback(self.stop)
-        return future
+    @staticmethod
+    def handle_async_exception(loop, context):
+        """Handle exceptions from async coroutines."""
+        _LOGGER.error("Caught exception")
+        _LOGGER.error(context)
```

Comment:
Do you think this logging error should be added into the same one?

Also, when I kill opsdroid this is being triggered by the `connector.listen` method (I'm assuming due to the infinite loop)

```
INFO opsdroid.core: Exiting application with return code 0
DEBUG asyncio: Using selector: KqueueSelector
ERROR opsdroid.core: Caught exception
ERROR opsdroid.core: {'message': 'Task was destroyed but it is pending!', 'task': <Task pending coro=<RocketChat.listen() running at /Users/fabiorosado/Documents/GitHub/opsdroid/opsdroid/connector/rocketchat/__init__.py:165> wait_for=<Future pending cb=[BaseSelectorEventLoop._sock_connect_done(10)(), <TaskWakeupMethWrapper object at 0x10e7a98e8>()]> cb=[gather.<locals>._done_callback(2)() at /usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/tasks.py:609]>}
ERROR opsdroid.core: Caught exception
ERROR opsdroid.core: {'message': 'Task was destroyed but it is pending!', 'task': <Task pending coro=<ConnectorShell.listen() running at /Users/fabiorosado/Library/Application Support/opsdroid/opsdroid-modules/connector/shell/__init__.py:70> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x10e7a9978>()]> cb=[gather.<locals>._done_callback(3)() at /usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/tasks.py:609]>}
```


This is probbaly outside the scope of this PR and more related to #247

Reply:
The two connectors which are complaining `rocketchat` and `shell` are not honoring the `disconnect` method. Therefore I would expect to see this error. 

Happy to make it one line though.

Reply:
I gave this another thought and I think two line its fine as it makes it easier to fix the issue - it might add a massive block of text is both of them are in one line.  I also looked at rocket.chat and I kind off forgot to add that haha